Renames `number` to `number_of_trials`

jeppe-style opened the pull request:
changes `configuration.termination_criteria.experiment.number` to `configuration.termination_criteria.experiment.number_of_trials`

depends on https://github.com/benchflow/benchflow/pull/459

VincenzoFerme reviewed (commented):
The PR has a commit from another PR: 2d36e7f

Maybe the issue is that you force pushed #419

I Reviewed it, when finalised we need to clean it up before merging.

VincenzoFerme committed:
Renames `number` to `number_of_trials` in experiment termination criteria

# Conflicts:
#	benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/terminationcriteria/experiment/ExperimentTerminationCriteria.scala
#	benchflow-dsl/src/test/scala/cloud/benchflow/dsl/definition/configuration/terminationcriteria/experiment/ExperimentTerminationCriteriaTest.scala

VincenzoFerme committed:
Merge branch 'devel' into feature-dsl-experiment-termination-criteria-renaming
